app/controllers/backgrounds_controller.rb

```diff
@@ -26,16 +30,32 @@ def edit
   # POST /backgrounds
   # POST /backgrounds.json
   def create
-    @background = Background.new(background_params)
+    Rails.logger.info("Initiate background creation: #{params}")
+    name = params[:name]
+    description = params[:description]
+    sample_ids = params[:sample_ids]
 
-    respond_to do |format|
-      if @background.save
-        format.html { redirect_to @background, notice: 'Background was successfully created.' }
-        format.json { render :show, status: :created, location: @background }
-      else
-        format.html { render :new }
-        format.json { render json: @background.errors, status: :unprocessable_entity }
-      end
+    if sample_ids.any? { |sid| !current_power.samples.pluck(:id).include?(sid) }
```

Comment:
Shouldn't these checks be done in the model ?

Reply:
Yes, but the background creation is asynchronous. We need to let the user know what's wrong as soon as they submit their request